src/Evolution/Systems/CurvedScalarWave/Characteristics.cpp

```diff
@@ -152,11 +153,15 @@ evolved_fields_from_characteristic_fields(
 
 template <size_t SpatialDim>
 double ComputeLargestCharacteristicSpeed<SpatialDim>::apply(
-    const std::array<DataVector, 4>& char_speeds) noexcept {
-  std::array<double, 4> max_speeds{
-      {max(abs(char_speeds[0])), max(abs(char_speeds[1])),
-       max(abs(char_speeds[2])), max(abs(char_speeds[3]))}};
-  return *std::max_element(max_speeds.begin(), max_speeds.end());
+    const Scalar<DataVector>& gamma_1, const Scalar<DataVector>& lapse,
+    const tnsr::I<DataVector, SpatialDim, Frame::Inertial>& shift,
+    const tnsr::ii<DataVector, SpatialDim, Frame::Inertial>&
+        spatial_metric) noexcept {
+  const auto shift_magnitude = magnitude(shift, spatial_metric);
```

Comment:
I don't understand why the magnitude of the shift is correct, as opposed to the sum of the absolute value of the components, or the max of the absolute value of the components? This is probably a horrible failure of basic linear algebra on my part...

Reply:
The char speeds are `\propto n_k N^k = g_jk n^j N^k` which is maximized when the normal vector is parallel to the shift vector, i.e. `n^j = N^j / \sqrt{N^i N_i}`, which means the max speeds will be `\propto n_k N^k = g_jk N^j N^k / \sqrt{N^i N_i} = \sqrt{N^i N_i} = magnitude(shift, spatial_metric)`.

Reply:
Thanks :)

Reply:
added